modules/space/rendering/renderablefluxnodes.cpp

```diff
@@ -0,0 +1,1093 @@
+    _pEarthdistGroup.addProperty(_pEnhancemethod);
+    _pEarthdistGroup.addProperty(_pInterestingStreamsEnabled);
+
+    _pFlowGroup.addProperty(_pFlowEnabled);
+    _pFlowGroup.addProperty(_pFlowColor);
+    _pFlowGroup.addProperty(_pFlowParticleSize);
+    _pFlowGroup.addProperty(_pFlowParticleSpacing);
+    _pFlowGroup.addProperty(_pFlowSpeed);
+    _pFlowGroup.addProperty(_pUseFlowColor);
+
+    _pCameraPerspectiveGroup.addProperty(_pCameraPerspectiveEnabled);
+    _pCameraPerspectiveGroup.addProperty(_pPerspectiveDistanceFactor);
+    _pCameraPerspectiveGroup.addProperty(_pDrawingCircles);
+    _pCameraPerspectiveGroup.addProperty(_pDrawingHollow);
+    _pCameraPerspectiveGroup.addProperty(_pGaussianAlphaFilter);
+    _pCameraPerspectiveGroup.addProperty(_pRadiusPerspectiveEnabled);
+    _pCameraPerspectiveGroup.addProperty(_pMaxNodeSize);
+    _pCameraPerspectiveGroup.addProperty(_pMinNodeSize);
+    _pCameraPerspectiveGroup.addProperty(_pPulseEnabled);
+    _pCameraPerspectiveGroup.addProperty(_pGaussianPulseEnabled);
+    _pCameraPerspectiveGroup.addProperty(_pPulseAlways);
+
+    definePropertyCallbackFunctions();
+}
+
+void RenderableFluxNodes::deinitializeGL() {
+    glDeleteVertexArrays(1, &_vertexArrayObject);
+    _vertexArrayObject = 0;
+
+    glDeleteBuffers(1, &_vertexPositionBuffer);
+    _vertexPositionBuffer = 0;
+
+    glDeleteBuffers(1, &_vertexColorBuffer);
+    _vertexColorBuffer = 0;
+
+    glDeleteBuffers(1, &_vertexFilteringBuffer);
+    _vertexFilteringBuffer = 0;
+
+    if (_shaderProgram) {
+        global::renderEngine->removeRenderProgram(_shaderProgram.get());
+        _shaderProgram = nullptr;
+    }
+}
+
+bool RenderableFluxNodes::isReady() const {
+    return _shaderProgram != nullptr;
+}
+
+void RenderableFluxNodes::populateStartTimes() {
+
+    // number of  characters in UTC ISO8601 format (without additional Z)
+    // 'YYYY-MM-DDTHH-MM-SS-XXX'
+    constexpr const int timeFormatSize = 23;
+
+    std::string timeFile = "";
+    std::string fileType = "";
+    for (const std::string& filePath : _binarySourceFiles) {
+        timeFile = filePath;
+
+        if (filePath.substr(filePath.find_last_of(".") + 1) == "csv" ) {
+            fileType = "csv";
+            break;
+        }
+
+        else if (filePath.substr(filePath.find_last_of(".") + 1) == "dat") {
+            fileType = "dat";
+            break;
+        }
+
+        else if (filePath.substr(filePath.find_last_of(".") + 1) == "txt") {
+            fileType = "txt";
+            break;
+        }
+        //if no file extention but word "time" in file name
+        else if (filePath.find("time") != std::string::npos && 
+                    filePath.find(".") == std::string::npos) {
+            break;
+        }
+        else {
+            LERROR(fmt::format("Error in file type or nameing of file '{}'.",
+                " Time meta file supports csv, dat, txt or without file extention",
```

Comment:
In general i like to put the empty characters at the end of the previous line, but that is personal preference